qualtran/bloqs/arithmetic/subtraction.py

```diff
@@ -45,10 +47,9 @@
 class Subtract(Bloq):
     r"""An n-bit subtraction gate.
 
-    Implements $U|a\rangle|b\rangle \rightarrow |a\rangle|a-b\rangle$ using $4n - 4 T$ gates.
+    Implements $U|a\rangle|b\rangle \rightarrow |a\rangle|a-b\rangle$ using $4n - 4$ T gates.
 
-    This construction uses `XGate` and `AddK` to compute the twos-compliment of `b` before
-    doing a standard `Add`.
+    This construction uses the relation `a - b = ~(~a + b)` to turn the operation into addition.
```

Comment:
are there any quantum references where this construction is used?

Reply:
@tanujkhattar you mentioned that this trick is used in some phase gradient papers, right?

Reply:
From https://arxiv.org/pdf/2310.03011#page=174.23

<img width="1248" alt="image" src="https://github.com/quantumlib/Qualtran/assets/7863287/355c3a30-16f6-4b14-823e-16d5080161c6">